crates/egui-winit/src/lib.rs

```diff
@@ -820,11 +821,16 @@ impl State {
 
         if let Some(ime) = ime {
             let pixels_per_point = pixels_per_point(&self.egui_ctx, window);
-            let ime_rect_px = pixels_per_point * ime.rect;
-            if self.ime_rect_px != Some(ime_rect_px)
-                || self.egui_ctx.input(|i| !i.events.is_empty())
-            {
-                self.ime_rect_px = Some(ime_rect_px);
+            let ime_rect_px = ime.rect * pixels_per_point;
+            let mut need_set_ime_cursor_area = true;
+
+            // On Wayland of Linux, repaints every frame Issue : See https://github.com/emilk/egui/pull/4254
```

Comment:
Wasn't this fixed in https://github.com/emilk/egui/pull/4269 ?

Reply:
Since the problem occurs on Windows, we have moved it down to only apply on Linux.